arango/executor.py

```diff
@@ -275,19 +279,34 @@ def __init__(
         if max_size is not None:
             data["maxTransactionSize"] = max_size
 
-        request = Request(
-            method="post",
-            endpoint="/_api/transaction/begin",
-            data=data,
-            headers={"x-arango-allow-dirty-read": "true"} if allow_dirty_read else None,
-        )
-        resp = self._conn.send_request(request)
+        self._id: str = ""
+        if transaction_id is None:
+            request = Request(
+                method="post",
+                endpoint="/_api/transaction/begin",
+                data=data,
+                headers={"x-arango-allow-dirty-read": "true"}
+                if allow_dirty_read
+                else None,
+            )
+            resp = self._conn.send_request(request)
 
-        if not resp.is_success:
-            raise TransactionInitError(resp, request)
+            if not resp.is_success:
+                raise TransactionInitError(resp, request)
+
+            result = resp.body["result"]
+            self._id = result["id"]
+        else:
+            request = Request(
+                method="get",
+                endpoint=f"/_api/transaction/{transaction_id}",
+            )
+            resp = self._conn.send_request(request)
+
+            if not resp.is_success:
+                raise TransactionInitError(resp, request)
```

Comment:
I'm not convinced that it's a good idea to do this explicitly instead of calling `status()`, as the code is almost the same.

I imagine something along the lines of this:

```suggestion
            self._id = transaction_id
            self.status()
```

It's shorter and it shows your intention clearly - you do this stunt in order to check the transaction's status. If you feel like it's too short, just add a brief comment.

Note that `status()` throws `TransactionStatusError` instead of `TransactionInitError`. I have mixed feelings about this - you can choose to leave it like that, or catch it and re-throw a `TransactionInitError`, or even introduce a `TransactionFetchError`. I think all have pros/cons, depending on the user's interpretation. Feel free to do as you like regarding the throw (don't forget to update the test if you change the error).

Reply:
Oh yeah, that makes a lot more sense. I just didn't think of doing this!

Reply:
I think raising an explicit exception is the way to go. Is there any reason to raise it from the original exception, or do you think it's fine to just pass along the relevant parameters to our new exception?

Reply:
This is fine